src/rollup/widgets/actions/construct-widget-actions.js

```diff
@@ -234,35 +247,31 @@ export default function constructWidgetActions (widget, config) {
             event: event
           };
 
+          // include the data from the form if we have it.
           if (data) {
             param.data = data;
           }
 
-          config.confirmAction.callback(callbackTypes.MODAL_CONFIRM, param);
-        }
-      }
+          // if waitForAsyncResponse we will handle the states as part of the callback
+          if (config.confirmAction.waitForAsyncResponse === true) {
```

Comment:
I see that the outer check `==='object'`. Will it be a problem if it's an empty `{}`?

Reply:
I don't think so. If `config.confirmAction` is an empty object, then `config.confirmAction.waitForAsyncResponse` is undefined. This is essentially equivalent to doing:

```js
var t;
if (t === true) {
  // do something
}
```

And in that case `t === true` just evaluates to false.

Reply:
Ahhh yes.